jwplayer-appletv-web-app/js/playback/Playback.js

```diff
@@ -0,0 +1,180 @@
+    player.seek();
+  }
+
+  function pause() {
+    EventBus.publish(Events.PAUSE);
+    player.pause();
+  }
+
+  function stop() {
+    player.stop();
+  }
+
+  function present() {
+    player.present();
+  }
+
+  function setOverlay(overlayDocument) {
+    EventBus.publish(Events.PLAYER_OVERLAY, {
+      overlayDocument: overlayDocument
+    });
+    player.overlayDocument = overlayDocument;
+  }
+
+  /**
+   * Handles player stateDidChange event
+   */
+  function stateHandler(evt) {
+    //console.log('stateHandler %O', evt);
+    EventBus.publish(Events.PLAYER_STATE, {
+      state: evt.state,
+      oldState: evt.oldState
+    });
+    if (evt.state == 'begin') {
+      EventBus.publish(Events.PLAYLIST_START, {
+        mediaItem: player.currentMediaItem
+      });
+    }
+  }
+
+  /**
+   * Handles player timeDidChange event
+   */
+  function timeHandler(evt) {
+    var time = evt.time ? evt.time : evt.boundary;
+    if (time == player.currentMediaItem.duration) {
+      EventBus.publish(Events.MEDIA_COMPLETE, {
+        item: player.currentMediaItem
+      });
+    } else if (time) {
+      EventBus.publish(Events.MEDIA_TIME, {
+        time: time
+      });
+    }
+  }
+
+  /**
+   * Handles the mediaItemWillChange event.
+   */
+  function mediaItemWillChangeHandler(evt) {
+    //console.log('mediaItemWillChangeHandler %O', evt);
+    /*
+      Valid values for reason (TV JS docs are not correct):
+      -  0 (Unknown)
+      -  1 (Played to end)
+      -  2 (Forwarded to end)
+      -  3 (Errored)
+      -  4 (Playlist changed)
+      -  5 (User initiated)
+    */
+    EventBus.publish(Events.MEDIA_CHANGE, {
+      reason: evt.reason,
+      previousMediaItem: player.previousMediaItem,
+      nextMediaItem: player.currentMediaItem
+    });
+  }
+
+  /**
+   * Handles the mediaItemDidChange event.
+   */
+  function mediaItemChangedHandler(evt) {
+    //console.log('mediaItemChangedHandler %O', evt);
+    // Reregister timeBoundaryDidCross handler
+    player.removeEventListener("timeBoundaryDidCross", timeHandler);
+    if (player.currentMediaItem) {
+      player.addEventListener("timeBoundaryDidCross", timeHandler, [player.currentMediaItem.duration]);
+    }
+    EventBus.publish(Events.MEDIA_CHANGED, {
+      reason: evt.reason,
+      previousMediaItem: player.previousMediaItem,
+      currentMediaItem: player.currentMediaItem
+    });
+  }
+
+  /**
+   * Handles the requestSeekToTime event.
+   */
+  function requestSeekToTimeHandler(evt) {
+    //console.log('requestSeekToTimeHandler %O', evt);
+    EventBus.publish(Events.SEEK, { // TODO: Should be seeked? TV JS docs seem to be incorrect on this event.
+      currentTime: evt.currentTime,
+      requestedTime: evt.requestedTime
+    });
+  }
+
+  function timedMetadataHandler(evt) {
+    //console.log('timedMetadataHandler %O', evt);
+    // TODO: meta event
+  }
+
+  function shouldHandleStateChangeHandler(event) {
+    if (!player.currentMediaItem.duration) {
```

Comment:
Hmm, I saw it fire in other occasions where it did contain a valid duration. This is worth a little investigation.